go_scripts/demystifier.go

```diff
@@ -48,23 +48,23 @@ func PrintTestSummary(testData *demystifier.TestRunData) {
 	for i := range testData.TestRun {
 		var numAttempts, failedAttempts, numOver1Second int
 		totalRunTime := time.Duration(0)
-
-		for j := range testData.TestRun[i].Attempt {
+		thisTest := &testData.TestRun[i]
```

Comment:
The key is to define thisTest using an address from testData.TestRun slice, which would allow subsequent changes to thisTest to modify testData.

Previously in orignal implementation `thisTest` came from for loop range var like below, which without double pointer would be a copy, and does not link back to original testData.
```go
	for _, thisTest := range testData.TestRun {
```
Modifying `thisTest` does nothing to `testData.TestRun[i]` without double pointer.

Now, defining like below, `thisTest` would be linked to testData.TestRun slice and modifying `thisTest` will update testData.
```go
		thisTest := &testData.TestRun[i]
```